### ISSUE TITLE
"functions" does not match qualified function names

### ISSUE THREAD

**Issue by rcook** [ndmitchell/hlint#582]
Example rule in `.hlint.yaml`:

```yaml
- functions:
  - {name: Data.List.nub, within: []}
  - {name: L.nub, within: []}
  - {name: nub, within: []}
```

Example code:

```haskell
module Main (main) where

main :: IO ()
main = print $ nub [1, 1, 2, 2]
```

Running hlint on this program detects the use of the banned function `nub`. However, hlint fails to detect the use in the following program (which references `nub` using the qualified form `L.hub`):

```haskell
module Main (main) where

import qualified Data.List as L

main :: IO ()
main = print $ L.nub [1, 1, 2, 2]
```

I'm using version 2.1.9 of hlint:

```
$ hlint --version
HLint v2.1.9, (C) Neil Mitchell 2006-2018
```

Is this expected behaviour?

**Comment by ndmitchell** [comment]
Looking at the code, not implemented is probably more accurate than a bug - there is no attempt to match anything used any kind of qualified smarts at all. Restricting functions isn't going to be massively useful until that's implemented.

**Comment by ndmitchell** [comment]
Note you can still write a rule that replaces `nub` or `Data.List.nub` with `bad`, which does know about qualifications etc.
